File: Code/Tier3-DatabaseLayer/RestServer/Common/GroupMember.cs
﻿using System;

 namespace WebApplication.Common
{
    public class GroupMember
    {
        public User user { get; set; }
        public String role { get; set; }
        public int groupID { get; set; }
        public Boolean validated { get; set; }
        public String username { get; set; }

        public GroupMember(User user){
            this.user = user;
        }

        public GroupMember()
        {
            user = new User();
        }

        public GroupMember(string username, string groupRole, int groupId)
        {
            user.username = username;
            role = groupRole;
            groupID = groupId;
        }


    }
}

File: Code/Tier2/Common/src/main/java/javaModel/GroupMember.java
package javaModel;

import java.io.Serializable;

public class GroupMember implements Serializable {

    public User user;
    public String role;
    public int groupID;
    private boolean validated;


    public GroupMember () { }

    public boolean isValidated() {
        return validated;
    }

    public void setValidated(boolean validated) {
        this.validated = validated;
    }

    public GroupMember(User user, String role)
    {
        this.user = user;
        this.role = role;
        validated=false;
    }

    public User getUser(){
        return user;
    }

    public void setUser(User user){
        this.user = user;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public int getGroupID() {
        return groupID;
    }

    public void setGroupID(int groupID) {
        this.groupID = groupID;
    }

    }

